### PR TITLE
Sl 18732: Fix responsiveness of GLTF texture transform controls in build floater

### PR DESCRIPTION
**Pull request by cosmic-linden**

### CODE REVIEW

File: indra/newview/llgltfmateriallist.cpp [hunk 1]
```diff
@@ -287,15 +293,21 @@ class LLGLTFMaterialOverrideDispatchHandler : public LLDispatchHandler
                         }
                         else if (obj && obj->isAnySelected())
                         {
-                            LLMaterialEditor::updateLive(object_override.mObjectId, results[i].mSide);
+                            for (auto& override_update_callback : callbacks)
```
Comment (akleshchev) [inline indra/newview/llgltfmateriallist.cpp]: this probably should be a function, like makeCallbacks(object_override.mObjectId, results[i].mSide).

**Review by akleshchev — APPROVED** [review]

File: indra/newview/llgltfmateriallist.h [hunk 2]
```diff
@@ -74,6 +74,8 @@ class LLGLTFMaterialList
     // Automatically called once per frame, but may be called explicitly
     // for cases that care about the done_callback forwarded to LLCoros::instance().launch
     static void flushUpdates(void(*done_callback)(bool) = nullptr);
+
+    static void addUpdateCallback(void(*update_callback)(const LLUUID& object_id, S32 side));
```
Comment (akleshchev) [inline indra/newview/llgltfmateriallist.h:78]: This only works for objects with seelctions, so might be better to name it addSelectionUpdateCallback or in some other way that indicates that this isn't a blanket update.

**Review by brad-linden — APPROVED** [review]

File: indra/newview/llgltfmateriallist.cpp [hunk 3]
```diff
@@ -288,6 +295,10 @@ class LLGLTFMaterialOverrideDispatchHandler : public LLDispatchHandler
                         else if (obj && obj->isAnySelected())
```
Comment (akleshchev) [inline indra/newview/llgltfmateriallist.cpp:292]: Might be a good time to replace "obj->isAnySelected()" this with (getTE(results[i].mSide) && getTE(results[i].mSide)->isSelected()) to minimize update spam.

**Comment by cosmic-linden** [comment]
@akleshchev Good points, especially with regard to being specific to selections. I pushed changes in response to your feedback.

**Pull request merged**